### PR TITLE
Initialize integration adapter project

### PR DESCRIPTION
**Pull request by saradj411**
Sara Djuric:implement sftp,implement pdf generator feature
Kristina Mucibabic: implement rabbitMQ, create database
Verica Mucibabic:implement rabbitMQ, create database
Maja Milenkovic:implement pharmacy registration feature, implement
frontend
All: add unit tests, implement persistance entities

### CODE REVIEW

File: MQuince/MQuince.Integration.Services.Implementation/MedicationsConsumptationService.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                    listMedicationsByDate.Append(medication);
+                }
+            }
+            
+            return listMedicationsByDate;
+
+
+        }
+        
+        public void GeneratePdf(DateDTO dto)
```
Comment (dusancina98) [inline MQuince/MQuince.Integration.Services.Implementation/MedicationsConsumptationService.cs:58]: This method could be refactored. Creating a new Class (eg. PdfGenerator) with couple of methods should do the job. But this is a lot better then in last commit.

File: MQuince/MQuince.Integration.Services.Implementation/SftpService.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace MQuince.Integration.Services.Implementation
+{
+    public class SftpService : ISftpService
+    {
+        public bool SendFile(string fileName)
```
Comment (dusancina98) [inline MQuince/MQuince.Integration.Services.Implementation/SftpService.cs:13]: This is a good place for refactoring in couple of methods. 
Throwing exception when client is not connected and handling in controller is better way of telling if the operation was succesfull. Then you can change return type to void.

File: MQuince/MQuince.Integration.HospitalApp/Startup.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            DbContextOptionsBuilder dbContextOptionsBuilder = new DbContextOptionsBuilder();
+            dbContextOptionsBuilder.UseMySql(@"server=localhost;user=root;password=root;database=pharmacydb");
```
Comment (dusancina98) [inline MQuince/MQuince.Integration.HospitalApp/Startup.cs:40]: Take a look at the MQuince.Application project from web team and appsettings.json from MQuince.WebAPI project.
You can make it more configurable by storing connection string in configuration file.

**Review by dusancina98 — APPROVED** [review]
A few more places for future refactor, but it's a lot better than last commit.

File: MQuince/MQuince.Integration.HospitalApp.Tests/MedicationsConsumptationMapperTest.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        }
+        [Fact]
+        public void MapMedicationsConsumptationPersistenceToMedicationsConsumptationEntity_Given_some_value_should_back_new_MyPharamcy()
+        {
+
+            var medicationsTest = new MedicationsConsumptionPersistance(new Guid(), "LekTest", new DateTime(2020, 11, 27), 2);
+
+            MedicationsConsumption medicationsConsumptation = MedicationsConsumptionMapper.MapMedicationsConsumptionPersistenceToMedicationsConsumptionEntity(medicationsTest);
+
+            Assert.NotNull(medicationsConsumptation);
```
Comment (zindovicstefan) [inline MQuince/MQuince.Integration.HospitalApp.Tests/MedicationsConsumptationMapperTest.cs]: Assert should check if each property is mapped correctly, to assert that return value of method is not null is not enough

File: MQuince/MQuince.Integration.HospitalApp.Tests/MedicationsConsumptationMapperTest.cs [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        }
+
+        [Fact]
+        public void MedicationsConsumptationEntityToMedicationsConsumptationPersistence_Given_some_value_should_back_new_MedicationsConsumptation()
+        {
+            MedicationsConsumption medicationsTest = new MedicationsConsumption(new Guid(), "LekTest", new DateTime(2020, 11, 27), 2);
+
+            MedicationsConsumptionPersistance persistance = MedicationsConsumptionMapper.MapMedicationsConsumptionEntityToMedicationsConsumptionPersistance(medicationsTest);
+
+            Assert.NotNull(persistance);
```
Comment (zindovicstefan) [inline MQuince/MQuince.Integration.HospitalApp.Tests/MedicationsConsumptationMapperTest.cs]: Assert should check if each property is mapped correctly, to assert that return value of method is not null is not enough

File: MQuince/MQuince.Integration.HospitalApp.Tests/OperationsWithDatabaseTest.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+            var stubRepository = new Mock<IPharmacyRepository>();
+
+            PharmacyDTO pharmacy = new PharmacyDTO()
+            {
+                Name = "some_name"
+            };
+
+            PharmacyService service = new PharmacyService(stubRepository.Object);
+
+            Assert.NotNull(service.Create(pharmacy));
```
Comment (zindovicstefan) [inline MQuince/MQuince.Integration.HospitalApp.Tests/OperationsWithDatabaseTest.cs]: Assert doesn't check if repository state is changed, just that returned value is not null

File: MQuince/MQuince.Integration.HospitalApp.Tests/PharmacyMapperTest.cs [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        }
+        [Fact]
+        public void MapPharmacyPersistenceToPharmacyEntity_Given_some_value_should_back_new_MyPharamcy()
+        {
+
+            var pharmacyTest = new PharmacyPersistence(new Guid(), "ApotekaTest", "apoteka.com");
+
+            MyPharmacy pharmacy = PharmacyMapper.MapPharmacyPersistenceToPharmacyEntity(pharmacyTest);
+
+            Assert.NotNull(pharmacy);
```
Comment (zindovicstefan) [inline MQuince/MQuince.Integration.HospitalApp.Tests/PharmacyMapperTest.cs]: Assert should check if each property is mapped correctly, to assert that return value of method is not null is not enough

File: MQuince/MQuince.Integration.HospitalApp.Tests/PharmacyMapperTest.cs [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+        [Fact]
+        public void MapPharmacyEntityToPharmacyPersistence_Given_some_value_should_back_new_MyPharamcy()
+        {
+            MyPharmacy pharmacy = new MyPharmacy(new Guid(), "ApotekaTest", "Url.com");
+
+            PharmacyPersistence pharmacyPersistance = PharmacyMapper.MapPharmacyEntityToPharmacyPersistence(pharmacy);
+
+            Assert.NotNull(pharmacyPersistance);
```
Comment (zindovicstefan) [inline MQuince/MQuince.Integration.HospitalApp.Tests/PharmacyMapperTest.cs]: Assert should check if each property is mapped correctly, to assert that return value of method is not null is not enough

File: MQuince/MQuince.Integration.HospitalApp/Controllers/MedicationsConsumptationController.cs [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+        public IEnumerable<IdentifiableDTO<MedicationsConsumptionDTO>> GetAll()
+        {
+            return _medicationsConsumptionService.GetAll();
+        }
+
+        [HttpPost]
+        public IActionResult Add([FromBody] MedicationsConsumptionDTO dto)
+        {
+            _medicationsConsumptionService.Create(dto);
+            return Ok(dto);
```
Comment (zindovicstefan) [inline MQuince/MQuince.Integration.HospitalApp/Controllers/MedicationsConsumptationController.cs]: http response code is not always 200 ok

File: MQuince/MQuince.Integration.HospitalApp/Controllers/PharmacyController.cs [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+        public IEnumerable<IdentifiableDTO<PharmacyDTO>> GetAll()
+        {
+            return _pharmacyService.GetAll();
+        }
+
+        [HttpPost]
+        public IActionResult Add(PharmacyDTO dto)
+        {
+            _pharmacyService.Create(dto);
+            return Ok(dto);
```
Comment (zindovicstefan) [inline MQuince/MQuince.Integration.HospitalApp/Controllers/PharmacyController.cs]: http response code is not always 200 ok

File: MQuince/MQuince.Integration.HospitalApp/RabbitMQService.cs [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MQuince.Integration.HospitalApp
+{
+    public class RabbitMQService : BackgroundService
```
Comment (zindovicstefan) [inline MQuince/MQuince.Integration.HospitalApp/RabbitMQService.cs:15]: This class should be refactored

File: MQuince/MQuince.Integration.Repository.MySQL/DataProvider/MedicationsConsumptionRepository.cs [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+            using (DataContext _context = new DataContext())
+            {
+                return MedicationsConsumptionMapper.MapMedicationsConsumptionPersistanceCollectionToMedicationsConsumptationEntityCollection(_context.MedicationsConsumptions.ToList());
+            }
+        }
+
+        public MedicationsConsumption GetById(Guid id)
+        {
+
+            throw new NotImplementedException();
```
Comment (zindovicstefan) [inline MQuince/MQuince.Integration.Repository.MySQL/DataProvider/MedicationsConsumptionRepository.cs]: Implement or remove method

File: MQuince/MQuince.Integration.Repository.MySQL/DataProvider/PharmacyRepository.cs [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+        {
+            using (DataContext _context = new DataContext())
+            {
+                return PharmacyMapper.MapPharmacyPersistenceCollectionToPharmacyEntityCollection(_context.Pharmacies.ToList());
+            }
+        }
+
+        public IEnumerable<MyPharmacy> GetByApi(bool publish)
+        {
+            throw new NotImplementedException();
```
Comment (zindovicstefan) [inline MQuince/MQuince.Integration.Repository.MySQL/DataProvider/PharmacyRepository.cs]: Implement or remove method

File: MQuince/MQuince.Integration.Services.Implementation/PharmacyService.cs [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+        public IEnumerable<IdentifiableDTO<PharmacyDTO>> GetByAllParams(bool publish, bool anonymous, bool approved)
+        {
+            throw new NotImplementedException();
+        }
+
+
+        public void Update(PharmacyDTO entityDTO, Guid id)
+        {
+            throw new NotImplementedException();
+        }
```
Comment (zindovicstefan) [inline MQuince/MQuince.Integration.Services.Implementation/PharmacyService.cs]: Implement or remove methods

**Review by zindovicstefan — CHANGES_REQUESTED** [review]
Commented regions should be resolved before PR is approved

**Review by zindovicstefan — APPROVED** [review]
All commented flaws are successfully resolved. All features from now on should be implemented using specified git workflow and git commit message format.

**Pull request merged**